src/assets/rewardrequestdb.cpp

```diff
@@ -0,0 +1,282 @@
+        __func__,
+        p_rewardID.c_str());
+
+    CRewardRequest rewardToRemove;
+    if (!RetrieveRewardWithID(p_rewardID, rewardToRemove)) {
+        LogPrintf("%s : Failed to find reward: rewardID='%s'!\n",
+            __func__,
+            p_rewardID.c_str());
+
+        return false;
+    }
+    //  Retrieve height-based entry which contains list of rewards to pay at that height
+    CRewardRequestDBEntry reqDbEntry;
+
+    //  If this fails, bail, since we can't be sure we won't blow away valid entries
+    if (!Read(std::make_pair(SCHEDULEDREWARD_FLAG, rewardToRemove.heightForPayout), reqDbEntry)) {
+        LogPrintf("%s : Failed to read scheduled rewards at specified height!\n", __func__);
+
+        return false;
+    }
+
+    //  If the list is empty, bail
+    if (reqDbEntry.requests.size() == 0) {
+        LogPrintf("%s : No scheduled reward exists at specified height!\n", __func__);
+
+        return true;
+    }
+
+    //  Remove this entry from the list
+    std::set<CRewardRequest>::iterator entryIT;
+
+    for (entryIT = reqDbEntry.requests.begin(); entryIT != reqDbEntry.requests.end(); ) {
+        if (entryIT->tgtAssetName == rewardToRemove.tgtAssetName) {
+            LogPrintf("%s : Removing reward: rewardID='%s', wallet='%s', height=%d, amt=%lld, srcAsset='%s', tgtAsset='%s', exceptions='%s'\n",
+                __func__,
+                rewardToRemove.rewardID.c_str(),
+                rewardToRemove.walletName.c_str(), rewardToRemove.heightForPayout, static_cast<long long>(rewardToRemove.totalPayoutAmt),
+                rewardToRemove.payoutSrc.c_str(), rewardToRemove.tgtAssetName.c_str(),
+                rewardToRemove.exceptionAddresses.c_str());
+
+            entryIT = reqDbEntry.requests.erase(entryIT);
+        }
+        else {
+            ++entryIT;
+        }
+    }
+
+    //  If the list still has entries, overwrite the record in the database
+    if (reqDbEntry.requests.size() > 0) {
+        bool succeeded = Write(std::make_pair(SCHEDULEDREWARD_FLAG, rewardToRemove.heightForPayout), reqDbEntry);
+
+        LogPrintf("%s : Removal of scheduled reward %s!\n",
+            __func__,
+            succeeded ? "succeeded" : "failed");
+
+        return succeeded;
+    }
+
+    //  Otherwise, erase the entire entry since none are left.
+    bool succeeded = Erase(std::make_pair(SCHEDULEDREWARD_FLAG, rewardToRemove.heightForPayout), true);
+
+    LogPrintf("%s : Removal of last scheduled reward %s!\n",
+        __func__,
+        succeeded ? "succeeded" : "failed");
+
+    return succeeded;
+}
+
+bool CRewardRequestDB::AreRewardsScheduledForHeight(const int & p_maxBlockHeight)
+{
+    LogPrintf("%s : Looking for scheduled rewards at height %d!\n",
+        __func__, p_maxBlockHeight);
+
+    std::unique_ptr<CDBIterator> pcursor(NewIterator());
+
+    pcursor->SeekToFirst();
+
+    // Find out if any pending rewards exist at or below the specified height
+    while (pcursor->Valid()) {
```

Comment:
or below? Doesn't look like this covers that

Reply:
Fixed comment by removing the "or below" portion, which was no longer applicable.